data-prepper-plugins/http-source/src/main/java/org/opensearch/dataprepper/plugins/source/loghttp/codec/JsonCodec.java

```diff
@@ -34,6 +34,27 @@ public List<String> parse(HttpData httpData) throws IOException {
             jsonList.add(recordString);
         }
 
+        return List.of(jsonList);
+    }
+
+    @Override
+    public List<List<String>> parse(HttpData httpData, int maxSize) throws IOException {
+        List<List<String>> jsonList = new ArrayList<>();
+        final List<Map<String, Object>> logList = mapper.readValue(httpData.toInputStream(),
+                LIST_OF_MAP_TYPE_REFERENCE);
+        int size = 2; // To account for "[" and "]" when the list is converted to String
```

Comment:
You can make this a constant string and include that data here.

```
private static final OVERHEAD_CHARACTERS = "[]"
```

and then:

```
int size = OVERHEAD_CHARACTERS.length()
```